src/watertap_contrib/reflo/unit_models/multi_effect_crystallizer.py

```diff
@@ -0,0 +1,335 @@
+                    - effect.temperature_operating,
+                    doc=f"Change in temperature at inlet for effect {n}",
+                )
+                effect.add_component(
+                    f"eq_delta_temperature_inlet_effect_{n}", del_temp_in_constr
+                )
+
+                del_temp_out_constr = Constraint(
+                    expr=effect.delta_temperature_out[0]
+                    == prev_effect.properties_pure_water[0].temperature
+                    - effect.properties_in[0].temperature,
+                    doc=f"Change in temperature at outlet for effect {n}",
+                )
+                effect.add_component(
+                    f"eq_delta_temperature_outlet_effect_{n}", del_temp_out_constr
+                )
+
+                hx_constr = Constraint(
+                    expr=prev_effect.energy_flow_superheated_vapor
+                    == effect.overall_heat_transfer_coefficient
+                    * effect.heat_exchanger_area
+                    * effect.delta_temperature[0],
+                    doc=f"Heat transfer equation for effect {n}",
+                )
+                effect.add_component(f"eq_heat_transfer_effect_{n}", hx_constr)
+
+                energy_flow_constr = Constraint(
+                    expr=effect.work_mechanical[0]
+                    == prev_effect.energy_flow_superheated_vapor,
+                    doc=f"Energy supplied to effect {n}",
+                )
+                effect.add_component(
+                    f"eq_energy_for_effect_{n}_from_effect_{n - 1}", energy_flow_constr
+                )
+
+        self.total_flow_vol_in = Expression(expr=total_flow_vol_in_expr)
+
+    def initialize_build(
+        self,
+        state_args=None,
+        outlvl=idaeslog.NOTSET,
+        solver=None,
+        optarg=None,
+    ):
+        """
+        Keyword Arguments:
+            state_args : a dict of arguments to be passed to the property
+                         package(s) to provide an initial state for
+                         initialization (see documentation of the specific
+                         property package) (default = {}).
+            outlvl : sets output level of initialization routine
+            optarg : solver options dictionary object (default=None)
+            solver : str indicating which solver to use during
+                     initialization (default = None)
+
+        Returns: None
+        """
+
+        init_args = dict(
+            state_args=state_args,
+            outlvl=outlvl,
+            solver=solver,
+            optarg=optarg,
+        )
+
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(self.name, outlvl, tag="unit")
+
+        opt = get_solver(solver, optarg)
+        for n, eff in self.effects.items():
+            if n == 1:
+                assert degrees_of_freedom(eff.effect) == 0
```

Comment:
This should raise an exception if DOF do not equal 0

Reply:
addressed